### ISSUE TITLE
defer: consider reporting exit like in deep-exit

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1300]
We have `deep-exit` rule that reports issues when using

We also have `defer` rule that reports bad usage of defer.

Both of them are useful, some people are using only one, other use

I feel like the code used in deep-exit rule could be reused in defer rule to report the defer that are broken.

The issue I see now with revive is that a code like this is not reported.

```go
 func main() {
     defer func() {
         fmt.Println("whatever")
     }()
 
     err := errors.New("test")
     if err != nil {
         log.Fatal("anything", err)
     }
 }
```

I do use defer and deep-exit, deep-exit reports the issues with the usage of exiter out of `main`

It would be something comparable to go-critic rule

https://go-critic.com/overview.html#exitafterdefer

**Comment by ccoVeille** [comment]
I discussed shortly with @alexandear before opening the issue to validate the idea. Let's talk now.

I don't know if it would be better to have an option in defer or not, if the option should be enabled by default.

It's open to debate
